honeycombio/resource_slack_recipient.go

```diff
@@ -0,0 +1,94 @@
+package honeycombio
+
+import (
+	"context"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	honeycombio "github.com/honeycombio/terraform-provider-honeycombio/client"
+)
+
+func newSlackRecipient() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceSlackRecipientCreate,
+		ReadContext:   resourceSlackRecipientRead,
+		UpdateContext: resourceSlackRecipientUpdate,
+		DeleteContext: resourceSlackRecipientDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Description: "Honeycomb Slack Recipient allows you to define and manage a Slack channel or user recipient that can be used by Triggers or BurnAlerts notifications.",
+
+		Schema: map[string]*schema.Schema{
+			"channel": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "The Slack channel or username to send the notification to. Must begin with `#` or `@`.",
+				ValidateFunc: validation.StringMatch(regexp.MustCompile(`^(#|@).+`), "channel must begin with `#` or `@`"),
+			},
+		},
+	}
+}
+
+func resourceSlackRecipientCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
It's these CRUD functions that all look really similar to me. Can they be cleaned up at all, or is this just how it is?